app/models/application_record.rb

```diff
@@ -20,4 +20,31 @@ def self.estimated_count
     result.clear # PG::Result is manually managed in memory, we need to release its resources
     count
   end
+
+  # Decorate object with appropriate decorator
+  def decorate
+    self.class.decorator_class.new(self)
+  end
+
+  def decorated?
+    false
+  end
+
+  # Decorate collection with appropriate decorator
+  def self.decorate
+    decorator_class.decorate_collection(all)
+  end
+
+  # Infers the decorator class to be used by (e.g. `User` maps to `UserDecorator`).
+  # adapted from https://github.com/drapergem/draper/blob/157eb955072a941e6455e0121fca09a989fcbc21/lib/draper/decoratable.rb#L71
+  def self.decorator_class(called_on = self)
+    prefix = respond_to?(:model_name) ? model_name : name
+    decorator_name = "#{prefix}Decorator"
+    decorator_name_constant = decorator_name.safe_constantize
+    return decorator_name_constant unless decorator_name_constant.nil?
+
+    return superclass.decorator_class(called_on) if superclass.respond_to?(:decorator_class)
+
+    raise UninferrableDecoratorError, "Could not infer a decorator for #{called_on.class.name}."
+  end
```

Comment:
these are support methods for "decoratable" objects, as we use decorators only on models, I didn't put it in a separate module, let me know if you'd like me to move that out, Draper has a `Decoratable` module

Reply:
I don't think this is too bad if this is the only decorator support code that makes it into `ApplicationRecord`. If more methods appear in the future, it could be refactored then!

Reply:
I never was a big fan of `.decorate`, as it's really hard to see where each decorator is used without looking at context. At a previous company we had a policy of always explicitly wrapping. We also didn't want to tie decoration to models, they usually already do too much in Rails and we had service objects that returned POROs, which depending on context also got decorated. 

But I understand that that's outside the scope of this PR, so LGTM. Modules that only get included in a single place feel like "cleaning up" by brushing the mess off the table into the drawer anyway 😉